text/src/autogluon/text/text_prediction/predictor/predictor.py

```diff
@@ -0,0 +1,398 @@
+                       f'For loading the model, you can use'
+                       f' `predictor = TextPredictor.load({self._path})`')
+        self.save(self._path)
+        return self
+
+    def evaluate(self, data, metrics=None):
+        """ Report the predictive performance evaluated for a given dataset.
+
+        Parameters
+        ----------
+        data : str or :class:`TabularDataset` or `pandas.DataFrame`
+            This Dataset must also contain the label-column with the same column-name as specified during `fit()`.
+            If str is passed, `valid_data` will be loaded using the str value as the file path.
+        metrics : str or List[str] or None
+            Name of metric or a list of names of metrics to report.
+            If it is not given, we will return the score of the stored eval_metric.
+
+        Returns
+        -------
+        ret : a single number or a dict of metric --> metric scores
+            Output
+        """
+        return self._model.evaluate(data, metrics=metrics)
+
+    def predict(self, dataset, as_pandas=False):
+        """Get the prediction from
+
+        Returns
+        -------
+        output
+            Array of predictions. One element corresponds to the prediction value of one
+        as_pandas
+            Whether to convert the output to a pandas dataframe
+        """
+        assert self._model is not None, 'Model does not seem to have been constructed. Have you called fit(), or load()?'
+        output = self._model.predict(dataset)
+        if as_pandas:
+            output = pd.DataFrame({self.label: output})[self.label]
+        return output
+
+    def predict_proba(self, dataset, as_pandas=False):
+        """Predict the probability from the input
+
+        Parameters
+        ----------
+        dataset
+            The dataset
+        as_pandas
+            Whether to convert the output to pandas dataframe
+
+        Returns
+        -------
+        output
+            The output matrix
+        """
+        assert self._model is not None,\
+            'Model does not seem to have been constructed. ' \
+            'Have you called fit(), or load()?'
+        output = self._model.predict_proba(dataset)
+        if as_pandas:
+            output = pd.DataFrame(output)
+        return output
+
+    def extract_embedding(self, dataset, stochastic_chunk=None, num_repeat=None, as_pandas=False):
+        """Extract the feature from the neural network
+
+        Returns
+        -------
+        output
+        """
+        assert self._model is not None, 'Model does not seem to have been constructed. ' \
+                                        'Have you called fit(), or load()?'
+        output = self._model.extract_embedding(dataset,
+                                               stochastic_chunk=stochastic_chunk,
+                                               num_repeat=num_repeat)
+        if as_pandas:
+            output = pd.DataFrame({self.label: output})
```

Comment:
This is for extracting embedding of the features right, so why is `self.label` the name used here?  
And why would this embedding only be 1 column? Shouldn't it be many columns?

Reply:
Revised.